### PR TITLE
Add functionality to list clustertasks

### PR DESCRIPTION
**Pull request by onyiny-ang**
# Changes

Addresses #118: Adding functionality to list ClusterTasks

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

```
Added support for listing clustertasks with `tkn clustertask list`
```

### CODE REVIEW

File: pkg/cmd/clustertask/clustertask.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import (
+	"github.com/spf13/cobra"
+	"github.com/tektoncd/cli/pkg/cli"
+	"github.com/tektoncd/cli/pkg/flags"
+)
+
+func Command(p cli.Params) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "clustertask",
+		Aliases: []string{"c", "clustertasks"},
```
Comment (sthaha) [inline pkg/cmd/clustertask/clustertask.go]: How about ct instead of `c`

File: pkg/cmd/clustertask/list.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+}
+
+func printClustertaskListObj(w io.Writer, p cli.Params, f *cliopts.PrintFlags) error {
+	cs, err := p.Clients()
+	if err != nil {
+		return err
+	}
+
+	clustertasks, err := listAllClustertasks(cs.Tekton)
+
```
Comment (sthaha) [inline pkg/cmd/clustertask/list.go]: nit: may want to remove the blank line

File: pkg/cmd/clustertask/list.go [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+	return clustertasks, nil
+}
+
+func listClustertaskDetails(cs versioned.Interface) (*v1alpha1.ClusterTaskList, error) {
+	clustertasks, err := listAllClustertasks(cs)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(clustertasks.Items) == 0 {
```
Comment (sthaha) [inline pkg/cmd/clustertask/list.go]: isn't this `if` superfluous? i.e. wouldn't the output be the same if we deleted it?

File: pkg/cmd/clustertask/list.go [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+)
+
+func listCommand(p cli.Params) *cobra.Command {
+	f := cliopts.NewPrintFlags("list")
+
+	c := &cobra.Command{
+		Use:     "list",
+		Aliases: []string{"ls"},
+		Short:   "Lists clustertasks in a namespace",
+		Long:    ``,
```
Comment (vdemeester) [inline pkg/cmd/clustertask/list.go]: We don't need to add the `Long` field if it's empty :stuck_out_tongue_closed_eyes: (we may need to fix that on other commands)

Comment (onyiny-ang) [reply pkg/cmd/clustertask/list.go]: Fixed and also deleted the "Long" flag from pipelines and task `list.go` `

File: pkg/cmd/clustertask/list.go (hunk 3 above)
Comment (onyiny-ang) [reply pkg/cmd/clustertask/list.go]: Removed from this and task `list.go` in follow up commit

File: pkg/cmd/clustertask/list.go [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+	// tektoncd go client fails to set these; probably a bug
+	clustertasks.GetObjectKind().SetGroupVersionKind(
+		schema.GroupVersionKind{
+			Version: "tekton.dev/v1alpha1",
+			Kind:    "ClustertaskList",
+		})
+	return clustertasks, nil
+}
+
+func listClustertaskDetails(cs versioned.Interface) (*v1alpha1.ClusterTaskList, error) {
```
Comment (sthaha) [inline pkg/cmd/clustertask/list.go]: Could you please help me understand why we need this func ? isn't this the same as `listAllClusterTasks`?

Comment (onyiny-ang) [reply pkg/cmd/clustertask/list.go]: Addressed in #165

**Pull request merged**